nvflare/app_opt/xgboost/histogram_based_v2/adaptor_executor.py

```diff
@@ -0,0 +1,173 @@
+
+        Returns:
+            An XGBClientAdaptor object
+        """
+        engine = fl_ctx.get_engine()
+        return engine.get_component(self.adaptor_component_id)
+
+    def handle_event(self, event_type: str, fl_ctx: FLContext):
+        if event_type == EventType.START_RUN:
+            adaptor = self.get_adaptor(fl_ctx)
+            if not adaptor:
+                self.system_panic(f"cannot get component for {self.adaptor_component_id}", fl_ctx)
+                return
+
+            if not isinstance(adaptor, XGBClientAdaptor):
+                self.system_panic(
+                    f"invalid component '{self.adaptor_component_id}': expect {XGBClientAdaptor.__name__} but got {type(adaptor)}",
+                    fl_ctx,
+                )
+                return
+
+            adaptor.set_abort_signal(self.abort_signal)
+            engine = fl_ctx.get_engine()
+            adaptor.set_sender(Sender(engine, self.req_timeout))
+            adaptor.initialize(fl_ctx)
+            self.adaptor = adaptor
+        elif event_type == EventType.END_RUN:
+            self.abort_signal.trigger(True)
+
+    def execute(self, task_name: str, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal) -> Shareable:
+        if task_name == self.configure_task_name:
+            # there are two important config params for the client:
+            #   the rank assigned to the client;
+            #   number of rounds for training.
+            ranks = shareable.get(Constant.CONF_KEY_CLIENT_RANKS)
+            if not ranks:
+                self.log_error(fl_ctx, f"missing {Constant.CONF_KEY_CLIENT_RANKS} from config")
+                return make_reply(ReturnCode.BAD_TASK_DATA)
+
+            if not isinstance(ranks, dict):
+                self.log_error(fl_ctx, f"expect config data to be dict but got {ranks}")
+                return make_reply(ReturnCode.BAD_TASK_DATA)
+
+            me = fl_ctx.get_identity_name()
+            my_rank = ranks.get(me)
+            if my_rank is None:
+                self.log_error(fl_ctx, f"missing rank for me ({me}) in config data")
+                return make_reply(ReturnCode.BAD_TASK_DATA)
+
+            self.log_info(fl_ctx, f"got my rank: {my_rank}")
+
+            num_rounds = shareable.get(Constant.CONF_KEY_NUM_ROUNDS)
+            if not num_rounds:
+                self.log_error(fl_ctx, f"missing {Constant.CONF_KEY_NUM_ROUNDS} from config")
+                return make_reply(ReturnCode.BAD_TASK_DATA)
+
+            secure = shareable.get(Constant.CONF_KEY_SECURE, False)
+
+            world_size = len(ranks)
+
+            # configure the XGB client target via the adaptor
+            self.adaptor.configure(
+                {
+                    Constant.CONF_KEY_RANK: my_rank,
+                    Constant.CONF_KEY_NUM_ROUNDS: num_rounds,
+                    Constant.CONF_KEY_WORLD_SIZE: world_size,
+                    Constant.CONF_KEY_SECURE: secure,
```

Comment:
I don't think we need "secure", since XGB will be deployed either in the same process or on the same host.

Reply:
right, I forgot to remove them, let me remove them.